builtin/common/misc_helpers.lua

```diff
@@ -250,11 +250,12 @@ local formspec_escapes = {
 	["["] = "\\[",
 	["]"] = "\\]",
 	[";"] = "\\;",
-	[","] = "\\,"
+	[","] = "\\,",
+	["$"] = "\\$",
```

Comment:
Only `${` needs escaping. What if you put `\\$` in a regular textarea field?

Reply:
I tested it and the escape character is just removed, so unnecessary escapes are harmless.